mlflow/store/artifact/artifact_repo.py

```diff
@@ -244,6 +246,58 @@ def max_workers(self) -> int:
         return min(num_cpus * _NUM_MAX_THREADS_PER_CPU, _NUM_MAX_THREADS)
 
 
+class MultipartUploadMixin(ABC):
+    @abstractmethod
+    def create_multipart_upload(
+        self, local_file: str, num_parts: int, artifact_path=None
+    ) -> CreateMultipartUploadResponse:
+        """
+        Initiate a multipart upload and retrieve the pre-signed upload URLS and upload id.
+
+        :param local_file: Path of artifact to upload
+        :param num_parts: Number of parts to upload. Only required by S3 and GCS.
+        :param artifact_path: Directory within the run's artifact directory in which to upload the
+                              artifact.
+        """
+        pass
+
+    @abstractmethod
+    def complete_multipart_upload(
+        self,
+        local_file: str,
+        upload_id: str,
+        parts: List,
```

Comment:
```suggestion
        parts: List[...],
```

Can we add the element type?